### ISSUE TITLE
refac

### ISSUE THREAD

**Issue by bart6114** [datarootsio/ml-skeleton-py#8]
- [x] remove poetry in favour of virtualenv
- [x] review unittests
- [x] remove dependency on click and use argparser
- [x] remove stuff that is not api focused (e.g. scripts/predict.py)
- [x] remove unnecessary still - make it as lightweight as possible - KISS (e.g. remove helper.py)
- [x] review etl/train/predict functions (simpler == better)
- [x] review docker files
- [x] review documentation README + HOWTO
- [x] add relevant badges (black, pr's welcome)

Baturay's Edit
- [x] remove gitlab-ci
- [x] simplify settings.py / the locations should work in all OS (e.g the prev version doesn't work in windows os.)
- [x] remove imblearn dependency, too specific and seems like outdated https://pypi.org/project/imblearn/
- [x] test set is generated but never used. Remove test set generation. (Use only train/validation set for simplicity)
- [x] Switch from pickle to joblib since we can compress the serialized file significantly in joblib.
- [x] Dploy-Kickstart integration, refac predict endpoint
- [x] implement serve.Dockerfile for dploy-kickstart. Implement feature for `docker-compose up serve`
- [x] Add .flake8 file to solve conflicts between flake8 and black. Flake8 template is exactly same with dploy-kickstart. (88 chars per line max.)
- [x] Document API

github ci workflow
- [x] run flake8 (with typing and docstrings)
- [x] run black
- [x] run unit tests
- [x] codecov

Tests Todos: (Bart, Batu, Gauthier  Discussion TBD)
- [x] E2E test
